rcl/src/rcl/rmw_implementation_identifier_check.c

```diff
@@ -116,6 +124,7 @@ rcl_ret_t rcl_rmw_implementation_identifier_check(void)
     // The strings at this point must be equal.
     // No need for asserted_rmw_impl anymore, free the memory.
     allocator.deallocate((char *)asserted_rmw_impl, allocator.state);
+    asserted_rmw_requires_free = false;
```

Comment:
instead of putting the flag back to `false` here, you can set `asserted_rmw_impl` to `NULL`.

Reply:
I didn't like the idea of the extra booleans, don't know how I didn't think this yesterday >,<